LiteCore/Database/PrebuiltCopier.cc

```diff
@@ -0,0 +1,84 @@
+//
+//  PrebuiltCopier.cc
+//  LiteCore
+//
+//  Created by Jim Borden on 2017/08/15.
+//  Copyright © 2017 Couchbase. All rights reserved.
+//
+
+#include "PrebuiltCopier.hh"
+#include "StringUtil.hh"
+#include "Error.hh"
+#include <thread>
+
+namespace litecore {
+    bool CopyPrebuiltDB(const litecore::FilePath &from, const litecore::FilePath &to,
+                             const C4DatabaseConfig *config, C4Error* outError) {
+        if(!from.exists()) {
+            Warn("No database exists at %s, cannot copy!", from.path().data());
+            outError->code = ENOENT;
+            outError->domain = POSIXDomain;
+            return false;
+        }
+        
+        FilePath backupPath;
+        Log("Copying prebuilt database from %s to %s", from.path().data(), to.path().data());
+        bool needBackup = to.exists();
+
+        try {
+            FilePath temp = FilePath::tempDirectory().mkTempDir();
+            temp.delRecursive();
+            from.copyTo(temp);
+            
+            auto db = c4db_open(c4str(temp.path().data()), config, outError);
+            if(db) {
+                Log("Resetting UUIDs...");
+                db->resetUUIDs();
+                c4db_free(db);
+                
+                if(needBackup) {
+                    Log("Backing up destination DB...");
+                    
+                    // TODO: Have a way to restore these temp backups if a crash happens?
+                    string p = to.path();
```

Comment:
`appendingToName` doesn't work here because it is a directory, and that function will append to the entire path, making `FOLDER/_TEMP` instead of `FOLDER_TEMP`